packages/block-editor/src/components/global-styles/utils.js

```diff
@@ -73,8 +73,20 @@ export const PRESET_METADATA = [
 	},
 	{
 		path: [ 'typography', 'fontSizes' ],
-		valueFunc: ( preset, { typography: typographySettings } ) =>
-			getTypographyFontSizeValue( preset, typographySettings ),
+		valueFunc: ( preset, settings ) =>
+			getTypographyFontSizeValue(
+				preset,
+				!! settings?.typography?.fluid && settings?.layout?.wideSize
```

Comment:
To avoid backwards compat problems I'm trying to keep the function signature the same, hence the gymnastics here.

Because we support `fluid: true` and `fluid: {...values}`, we have to ensure the param object is correctly set so that we can know if fluid typography is enabled or not.

Also, we need to pass `settings?.layout?.wideSize` if available only when fluid font sizes are enabled. 😵 

The alternative could be a third argument to the `getTypographyFontSizeValue()` function, e.g., `isFluidEnabled`. 

Or add a prop and pass the following second argument to `getTypographyFontSizeValue`:

```
{
    ....settings?.typography,
    layoutWideSize: settings?.layout?.wideSize
}

```

and move the logic there? I think that might be better.

Reply:
It is a bit of gymnastics to maintain back compat, but I actually don't mind it how it's written as-is, as it means that the calling code is responsible for ensuring that the spread `settings.typography.fluid` overrides the `maxViewPortWidth` set via the layout `wideSize`, and the `getTypographyFontSizeValue` doesn't have to work too hard to figure it out.

That said, adding `layoutWideSize` to `TypographySettings` could be a clever way to do it, then it'd be fairly straightforward to use `layoutWideSize` only if `maxViewPortWidth` doesn't exist.

Whatever you think works best, I don't think I mind either way, personally!

Reply:
Not all that familiar with this part of the codebase so take this with a large grain of salt.

What if we extract all of the code in `getTypographyFontSizeValue` that transforms the `settings.typography` object into an object that's suitable for passing to `getComputedFluidTypographyValue` into a new function?

That way `valueFunc` can set `maximumViewPortWidth` on a normalised object and not have to deal with the intricacies of parsing the `settings.typography.fluid` boolean / object.

```js
valueFunc: ( preset, settings ) => {
	const fluidTypographySettings = getFluidTypographySettings( settings?.typography );
	fluidSettings.maximumViewPortWidth = settings?.layout?.wideSize;
	return getComputedFluidTypographyValue( fluidSettings );
},
```

Reply:
I don't think what you have is that bad though. This suggestion is for if it's really bugging you 😀

Reply:
> What if we extract all of the code in getTypographyFontSizeValue that transforms the settings.typography object into an object that's suitable for passing to getComputedFluidTypographyValue into a new function?

yes

#### yes


### YES!


`packages/block-editor/src/components/global-styles/typography-utils.js` exists for a reason 😄

Reply:
> I don't think what you have is that bad though. This suggestion is for if it's really bugging you 😀

Thanks for helping @noisysocks and @andrewserong 

I might merge as is and mark [this suggestion ](https://github.com/WordPress/gutenberg/pull/51146#discussion_r1212602061)for a follow up, since it might come in handy down the track.